envsource_assign_test.go

```diff
@@ -0,0 +1,359 @@
+package staert
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/containous/flaeg/parse"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getPtrPtrConfig() *struct {
+	StringValue string
+	NextPointer **basicAppConfig
+} {
+	configPtr := &basicAppConfig{
+		BoolValue:   true,
+		IntValue:    1,
+		StringValue: "string",
+	}
+	expectedPtrPtr := &struct {
+		StringValue string
+		NextPointer **basicAppConfig
+	}{
+		StringValue: "FOO",
+	}
+	expectedPtrPtr.NextPointer = &configPtr
+	return expectedPtrPtr
+}
+
+func TestAssignValues(t *testing.T) {
+
+	parsers, _ := parse.LoadParsers(nil)
+	subject := &envSource{
+		"",
+		"_",
+		parsers,
+	}
+
+	expectedPtrPtr := getPtrPtrConfig()
+
+	testCases := []struct {
+		Label       string
```

Comment:
add the test cases description for each test.
use a field `desc`

Reply:
That's what the Label field should do. "BasicStructPointerPointer", "WithWrongPath". What format do you use for test names/descriptions?

Reply:
We use `desc` to avoid ambiguities of names like `name`, `label`, ...

The description must be a sentence or partial sentence.
ex: `with a wrong path`

I love the `should ... when ...` format.